### PR TITLE
Gun spawn overhaul

### PR DESCRIPTION
**Pull request by Holli-Git**
#### Summary
Content "Overhaul gun spawning"

#### Purpose of change

Gun spawns are all over the place. There's not much rhyme or reason why a gun would spawn in its quantity besides guesswork. This PR aims to bring gun spawns in-line with real-world numbers, so there's less of a headache when adding new guns. This information was based mostly on gunbroker results, and #37571.

#### Describe the solution
Uses gunbroker and #37571 to gauge how often a caliber should spawn, and how often a gun should spawn. This is by no means perfect and has lots of flaws, but currently its the best solution due to a lack of any other information. 

Removes itemgroups like rare/obscure guns and smg groups. This is because before, "rare guns" had a fixed chance to spawn somewhere, despite rarity not really working like that. Instead, guns now have to get a good dice roll to spawn somewhere, instead of it being fixed. SMG spawns were removed as a civilian can't really own an actual SMG, and so it was better split into either pistols or rifles.

Adjusts some behavior of conversion gun spawns, as they could spawn in guns you could not easily legally own.

#### Describe alternatives you've considered
Not doing this, it was a whole lot of work.

#### Testing
Loaded up a world before changes, loaded into the same one after without any errors. Tested a few itemgroups in game, such as the display ones to make sure no ammo spawned with it, etc. 

#### Additional context

Caused by #75925. Also requires a lot of reworking of guns, and ammo spawns. Yay.

### CODE REVIEW

**Review by GuardianDll — APPROVED** [review]
itemgroups looks okay

**Comment by tenehea** [comment]
> SMG spawns were removed as a civilian can't really own an actual SMG, and so it was better split into either pistols or rifles.

This is a very bizarre assertion, unless rifles contains semi-auto pistol caliber rifles.

**Comment by Holli-Git** [comment]
> This is a very bizarre assertion, unless rifles contains semi-auto pistol caliber rifles.

Yeah, they do now. Check the changes I've made

File: data/json/itemgroups/Weapons_Mods_Ammo/conversions/nested_conversion_guns.json [hunk 1]
```diff
@@ -1,83 +1,155 @@
...
+      { "group": "ar15_450medium", "charges": [ 0, 30 ] },
+      { "item": "450_ar15mag" },
+      { "item": "450_ar15mag", "prob": 50 },
+      { "group": "on_hand_450" }
+    ]
+  },
+  {
+    "id": "nested_ar15_retool_450bushmaster_7",
+    "type": "item_group",
+    "//": "This is a distribution for the gun, reasonable number of backup mags, and some ammo to repack.  7 inch rifle barrel, though keep in mind its a pistol.",
```
[the remark this answers is not in the document]
Comment (Holli-Git) [reply data/json/itemgroups/Weapons_Mods_Ammo/conversions/nested_conversion_guns.json]: Feels kinda pointless ngl but eh

**Comment by Night-Pryanik** [comment]
This is a big chunk of changes. It's really hard to review it properly (for me, at least). May I suggest splitting this PR into a smaller pieces, for example first one for pistols, second one for shotguns etc?

**Comment by Holli-Git** [comment]
> This is a big chunk of changes. It's really hard to review it properly (for me, at least). May I suggest splitting this PR into a smaller pieces, for example first one for pistols, second one for shotguns etc?

That would've been fairly harder to do, as since they cover a lot of the same areas it would introduce a lot of merge conflicts. Guns are also pretty interconnected, and I'm unsure if I could sort them into pistol/shotgun/rifle. Though I do understand where you're coming from, and am thankful for the review, so in the future I'll split it up into different PRs.

**Comment by kevingranade** [comment]
Ran my demographics test on top of this to see if anything skewed badly.  tl;dr nothing horrific, rifles vs pistols is kind of out of whack, but my source for that is pretty sketchy to start with, and I don't know if this PR did it or if it was already there.  I'll rerun on before-this-change to check, but it takes 8h to run so I'll have results tomorrow.


| gun type | expected | actual |
|--------|----------|-------|
| pistols | 0.616f | 0.47416f |
| rifles | 0.357f | 0.50239f |
| shotguns | 0.027f | 0.02345f |

| calibre | expected | actual |
|---------|----------|--------|
| 9mm | 0.21502f | 0.24853f |
| 223 | 0.1286f | 0.10938f |
| 45 | 0.05041f | 0.06391f |
| 380 | 0.04938f | 0.04019f |
| 40 | 0.04115f | 0.04782f |
| 357mag | 0.03498f | 0.0169f |
| 762 | 0.03498f | 0.02165f |
| 308 | 0.02572f | 0.08697f |
| 38 | 0.01955f | 0.02545f |
| 44 | 0.01543f | 0.00827f |
| 3006 | 0.00823f | 0.01782f |
| 45colt | 0.00823f | 0.00413f |
| 762R | 0.00412f | 0.00629f |
| 270win | 0.00309f | 0.00441f |
| 410shot | 0.00309f | 0.00101f |
| 300blk | 0.00206f | 0.0f |
| 10mm | 0.00103f | 0.00154f |
| 123ln | 0.00103f | 0.00069f |
| other | 0.11728f | 0.06595f |

**Comment by kevingranade** [comment]
Belated rerun from before this PR:

| gun type | expected | actual |
|----------|----------|--------|
| pistols | 0.616f | 0.41017f |
| rifles | 0.357f | 0.55517f | 
| shotguns | 0.027f | 0.03466f |

In short, it was already skewed this direction, though this PR *might* have contributed a bit.

**Pull request merged**